core/trino-main/src/test/java/io/trino/sql/analyzer/TestAnalyzer.java

```diff
@@ -2508,6 +2512,22 @@ public void testShowCreateDuplicateNames()
                 .hasMessageContaining("Relation 'tpch.s1.duplicate_view' is a view, not a table");
     }
 
+    @Test
+    public void testAnalysisDuplicateNames()
+    {
+        // There is materialized view, view and table with "duplicate_materialized_view" name.
+        // Materialized view redirects to "t1"
+        Analysis analysis = analyze("SELECT * FROM duplicate_materialized_view");
+        TableHandle handle = getOnlyElement(analysis.getTables());
+        assertThat(((TestingTableHandle) handle.getConnectorHandle()).getTableName().getTableName()).isEqualTo("t1");
+
+        // There is view and table with "duplicate_view" name.
+        // View redirects to "s4"
+        analysis = analyze("SELECT * FROM duplicate_view");
+        handle = getOnlyElement(analysis.getTables());
+        assertThat(((TestingTableHandle) handle.getConnectorHandle()).getTableName().getTableName()).isEqualTo("s4");
```

Comment:
Would it be possible to test SHOW queries  and information_schema.tables, information_schema.columns as well?

Reply:
What do you mean?